CDCgov/prime-reportstream
# Experience/5709: Upload HL7 file via Validation Service UI

This PR adds the beginning stages of the Validation Service Prototype, focused on being able to send an HL7 through the User Interface.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
As an admin:
1. try to navigate to /validate
2. you should get a 404 page
3. go to /admin/features
4. fill in "validation-service" and save
5. now navigate to /validate - you should be in!
6. first sender: CSV - switch to a CSV sender such as All In One Health (all-in-one-health-ca)
7. send a valid file
8. this should look the same as before with the old uploader
9. second sender: switch to an HL7 sender such as Healthcare Integrations (hcintegrations)
10. navigate back to /validate
11. throw in a HL7 file (something with .hl7 extention, see ticket for samples)
12. have fun!

## Changes
- added /validate route behind feature flag and admin permissions
- added Validate page
- allowing both CSV and HL7 in file input
- validation check for filetype of CSV
- validation check for HL7 by file extension (that's all we can really do right now)
- upgraded cypress
- unit tests for waters endpoint
- super basic unit test for `<Validate/>` component

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #5709 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
